nats.go

```diff
@@ -3790,6 +3790,89 @@ func (m *Msg) RespondMsg(msg *Msg) error {
 	return nc.PublishMsg(msg)
 }
 
+func (m *Msg) isJSMsg() bool {
+	if m.Reply == "" {
+		return false
+	}
+
+	parts := strings.Split(m.Reply, ".")
+	if len(parts) == 8 && parts[0] == "$JS" && parts[1] != "ACK" {
+		return true
+	}
+
+	return false
+}
+
+var (
+	AckAck      = []byte("+ACK")
+	AckNak      = []byte("-NAK")
+	AckProgress = []byte("+WPI")
+	AckNext     = []byte("+NXT")
+	AckTerm     = []byte("+TERM")
+)
+
+// Ack acknowledges a JetStream messages received from a Consumer, indicating the message
+// should not be received again later
+func (m *Msg) Ack() error {
+	if !m.isJSMsg() {
+		return fmt.Errorf("not a JetStream originated message")
+	}
+
+	return m.Respond(AckAck)
+}
+
+// AckConfirm performs an Ack() and requests a confirmation from JetStream that the ack was received
+func (m *Msg) AckConfirm(nc *Conn, timeout time.Duration) (*Msg, error) {
+	if !m.isJSMsg() {
+		return nil, fmt.Errorf("not a JetStream originated message")
+	}
+
+	if nc == nil {
+		return nil, fmt.Errorf("invalid connection supplied")
+	}
+
+	return nc.Request(m.Reply, AckAck, timeout)
+}
+
+// AckNak acknowledges a JetStream message received from a Consumer, indicating that the message
+// is not completely processed and should be sent again later
+func (m *Msg) AckNak() error {
```

Comment:
Just Nak() I think. Also same variable option to wait on ack of ack.

Reply:
I made em all Ack* so they are easy to find in IDEs and obviously related, but sure msg has few methods so we could do that.

Reply:
@derekcollison how do these look? if good I'll add tests etc